x/csr/types/genesis_test.go

```diff
@@ -3,38 +3,80 @@ package types_test
 import (
 	"testing"
 
-	"github.com/stretchr/testify/require"
 	"github.com/Canto-Network/Canto/v2/x/csr/types"
+	"github.com/evmos/ethermint/tests"
+	"github.com/stretchr/testify/suite"
 )
 
-func TestGenesisState_Validate(t *testing.T) {
-    for _, tc := range []struct {
-    		desc          string
-    		genState      *types.GenesisState
-    		valid bool
-    } {
-        {
-            desc:     "default is valid",
-            genState: types.DefaultGenesis(),
-            valid:    true,
-        },
-        {
-            desc:     "valid genesis state",
-            genState: &types.GenesisState{
-            	
-                // this line is used by starport scaffolding # types/genesis/validField
-            },
-            valid:    true,
-        },
-        // this line is used by starport scaffolding # types/genesis/testcase
-    } {
-        t.Run(tc.desc, func(t *testing.T) {
-            err := tc.genState.Validate()
-            if tc.valid {
-                require.NoError(t, err)
-            } else {
-                require.Error(t, err)
-            }
-        })
-    }
-}
\ No newline at end of file
+type GensisStateSuite struct {
+	suite.Suite
+	params types.Params
+	csrs   []*types.CSR
+}
+
+func TestGenesisStateSuite(t *testing.T) {
+	suite.Run(t, new(GensisStateSuite))
+}
+
+func (suite *GensisStateSuite) SetupTest() {
+	suite.params = types.DefaultParams()
+
+	contracts := []string{tests.GenerateAddress().String(), tests.GenerateAddress().String(), tests.GenerateAddress().String()}
+	poolAddress := tests.GenerateAddress().String()
+	nft1 := types.NewCSRNFT(0, poolAddress)
+	nft2 := types.NewCSRNFT(1, poolAddress)
+
+	csrPool := types.CSRPool{
+		NftSupply:   2,
+		PoolAddress: poolAddress,
+		CsrNfts:     []*types.CSRNFT{&nft1, &nft2},
+	}
+	csr := types.CSR{
+		Deployer:  tests.GenerateAddress().String(),
+		Contracts: contracts,
+		CsrPool:   &csrPool,
+	}
+	suite.csrs = []*types.CSR{&csr, &csr}
+
+}
+
+// Test all of the genesis states, when empty and when not
+func (suite *GensisStateSuite) TestGenesisStateValidate() {
+	testCases := []struct {
+		desc     string
+		genState *types.GenesisState
+		valid    bool
+	}{
+		{
+			desc:     "Default genesis parameters are valid - pass",
+			genState: types.DefaultGenesis(),
+			valid:    true,
+		},
+		{
+			desc: "Duplicate pool in genesis state - fail",
+			genState: &types.GenesisState{
+				Params: suite.params,
+				Csrs:   suite.csrs,
```

Comment:
Add test case for duplicated nftIds both across pools and within the same pool

Reply:
removing these test cases

Reply:
per slack